crates/solvers/src/boundary/rate_limiter.rs

```diff
@@ -0,0 +1,58 @@
+use {
+    anyhow::Result,
+    shared::rate_limiter::{
+        RateLimiter as SharedRateLimiter,
+        RateLimiterError as SharedRateLimiterError,
+        RateLimitingStrategy as SharedRateLimitingStrategy,
+    },
+    std::{future::Future, time::Duration},
+    thiserror::Error,
+};
+
+pub struct RateLimiter {
+    inner: SharedRateLimiter,
```

Comment:
nit: I would prefer to have here
`inner: shared::RateLimiter`

Reply:
That actually `shared::rate_limiter::RateLimiter` which is quite long and will be used across across the code. That's why I decided to stick with the shorter versions.

Reply:
You should be able to export `RateLimiter` to `shared` with `pub use` and therefore skip `::rate_limiter`